artemis-commons/src/main/java/org/apache/activemq/artemis/api/core/SimpleString.java

```diff
@@ -281,6 +283,27 @@ public String toString() {
       return str;
    }
 
+   public String[] getPaths(final char separator) {
```

Comment:
this is probably nit picking... 

but if you called getPaths('.')

and then getPaths('/');


the second one would return the paths from the previous one.



Should we care about that? or just add docs about it?

Reply:
that is a valid observation and probably a bug. It comes down to how and where separators are configured. if they are server wide then it ok. let me revert. thanks!

Reply:
It is ok, the configuration of separator is part of the server/postoffice, it won't change for the lifetime of an instance. I will add a comment to that effect.

Reply:
@gtully is that ok if I amend your commit with the following java doc:

```java
   /** Notice this method will cache the result of getPaths, subsequent calls will return the same result even if you passed in a new separator. 
       On the broker usage, the same separator from a configuration, which will change only upon restart. If you use this method outside of this context beware with that caveat. */
   public String[] getPaths(final char separator) {
```

Reply:
rebased and added some javadoc to that getPaths

Reply:
darn, I needed a refresh. Either comment is fine :-)